source/basic.tex

```diff
@@ -1752,6 +1752,42 @@
 If that lookup finds nothing, it undergoes unqualified name lookup;
 in each case, only names
 that denote types or templates whose specializations are types are considered.
+\begin{example}
+\begin{codeblock}
+struct A {
+  using T = int;
+  operator int();
+  operator long();
+};
+
+template<class X, class T>
+int f() {
+  return X().operator T();      // searches for \tcode{T} in the scope of \tcode{X} first
+}
+
+int x = f<A, long>();           // calls \tcode{A::operator int}
+
+struct T1 { struct U { int i; }; };
+struct T2 { struct U { int i; }; };
+struct U1 {};
+struct U2 {};
+
+struct B {
+  using T = T1;
+  using U = U1;
+  operator U1 T1::*();
+  operator U1 T2::*();
+  operator U2 T1::*();
+  operator U2 T2::*();
+};
+
+void g() {
+  using T = T2;
+  using U = U2;
+  B().operator U T::*();        // calls \tcode{B::operator U1 T1::*}
```

Comment:
Mostly because these examples are long, could we just combine them—make one of `T` and `U` a template parameter of `g`?

Reply:
Removed.